src/util/code_util.cpp

```diff
@@ -0,0 +1,20 @@
+/*******************************************************************\
+
+Module: codet helper utilities
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include "code_util.h"
+
+code_assignt get_null_assignment(
+  const exprt &expr,
+  const pointer_typet &ptr_type,
+  const source_locationt &loc)
+{
+  null_pointer_exprt null_pointer_expr(ptr_type);
+  code_assignt code(expr, null_pointer_expr);
```

Comment:
While you're at it ...
1) I'd use `code_assignt code{expr, null_pointer_exprt{ptr_type}};`
2) It would be nice to have a `PRECONDITION(expr.type() == ptr_type);` - or even get rid of `ptr_type`, because it must be the same as `expr.type()`.

Reply:
I did 1) in a separate commit. As for 2), I'm guessing `pointer_type` could generally be more specific than `expr_type()` and I'm not sure if that could cause any problems with a null assignment. In any case, I'll raise another PR with Romain's suggestion of making a new constructor to replace the helper function, and not waste too much time on this PR until I have reviews on the alternative solution.

Reply:
A `code_assignt` where `.lhs().type() != .rhs().type()` violates invariants. You could still have that more-specific type, but should then add `typecast_exprt::conditional_cast` to make sure a typecast is added as if and when it is needed.